nautobot_chatops/workers/nautobot.py

```diff
@@ -1045,6 +1051,338 @@ def get_circuit_providers(dispatcher, *args):
+
+    # Confirm kwargs is valid JSON
+    json_args = {}
+    try:
+        if json_string_kwargs:
+            json_args = json.loads(json_string_kwargs)
+    except json.JSONDecodeError:
+        dispatcher.send_error(f"Invalid JSON-string, cannot decode: {json_string_kwargs}")
+        return (CommandStatusChoices.STATUS_FAILED, f"Invalid JSON-string, cannot decode: {json_string_kwargs}")
+
+    profile = False
+    if json_args.get("profile") and json_args["profile"] is True:
+        profile = True
+
+    # Get the job model instance using job name
+    try:
+        job_model = Job.objects.restrict(dispatcher.user, "view").get(name=job_name)
+    except Job.DoesNotExist:
+        dispatcher.send_error(f"Job {job_name} not found")
+        return (CommandStatusChoices.STATUS_FAILED, f'Job "{job_name}" was not found')
+
+    if not job_model.enabled:
+        dispatcher.send_error(f"The requested job {job_name} is not enabled")
+        return (CommandStatusChoices.STATUS_FAILED, f'Job "{job_name}" is not enabled')
+
+    form_class = get_job(job_model.class_path).as_form()
+
+    # Parse base form fields from job class
+    form_fields = []
+    for field_name, _ in form_class.base_fields.items():  # pylint: disable=unused-variable
+        if field_name.startswith("_"):
+            continue
+        form_fields.append(f"{field_name}")
+
+    # Basic logic check with what we know, we should expect init-job-form vs init-job to parse the same base fields
+    if len(form_fields) != len(args):
+        dispatcher.send_error("The form class fields and the passed init-jobs args do not match.")
+        return (
+            CommandStatusChoices.STATUS_FAILED,
+            "The form class fields and the passed init-jobs args do not match.",
+        )
+
+    form_item_kwargs = {}
+    for index, _ in enumerate(form_fields):  # pylint: disable=unused-variable
+        # Check request input (string-type) is also valid JSON
+        if args[index][0] == "{":
+            try:
+                json_arg = json.loads(args[index])
+                if not json_arg.get("id"):
+                    dispatcher.send_error("Form field arg is JSON dictionary, and has no `id` key.")
+                    return (
+                        CommandStatusChoices.STATUS_FAILED,
+                        "Form field arg is JSON dictionary, and has no `id` key.",
+                    )
+                form_item_kwargs[form_fields[index]] = json_arg.get("id")
+                continue
+            except json.JSONDecodeError:
+                form_item_kwargs[form_fields[index]] = args[index]
+                continue
+        form_item_kwargs[form_fields[index]] = args[index]
+
+    job_result = JobResult.enqueue_job(
+        job_model=job_model,
+        user=dispatcher.user,
+        profile=profile,
+        **form_item_kwargs,
+    )
+
+    # Wait on the job to finish
+    while job_result.status not in JobResultStatusChoices.READY_STATES:
```

Comment:
We may want something to drop out of this loop after enough time?

Reply:
> We may want something to drop out of this loop after enough time?

Addressed in [7470ec8](https://github.com/nautobot/nautobot-app-chatops/pull/270/commits/7470ec861458a86e523e452ede9a7ef5e13af6cd)